src/bin/pg_autoctl/service_keeper_init.c

```diff
@@ -32,6 +32,32 @@
 #include "string_utils.h"
 #include "supervisor.h"
 
+/*
+ * service_keeper_handle_dynamic enables or disables dynamic services based on
+ * its configuration.
+ *
+ * Currenty no such services are present in the configuration
+ *
+ * Returns the number of services, positive when added, negative when removed.
+ */
+static int
+service_keeper_handle_dynamic(Supervisor *supervisor, void *arg)
```

Comment:
Not really, because there can be more than one services added. The intension as per the discussion in the PR which this is cut from, is the services to be defined in the config file. As such the config file might have had 2 new dynamic services enabled and 3 old ones disabled. What is of interest to the supervisor is the net change.

Reply:
Okay then I would still insist on returning a bool. Then either have `int *diffCount` in the arguments or maybe expand fully with two arrays `addedServices[], removedServices[]` allocated by the caller, and that we fill-in with copies of the services discovered there.